src/boot/grub2/ostree-grub-generator

```diff
@@ -61,7 +61,12 @@ read_config()
 
 populate_menu()
 {
-    boot_prefix="${OSTREE_BOOT_PARTITION}"
+    # Default to /boot if OSTREE_BOOT_PARTITION is not set
```

Comment:
Hmm...I'm worried that some people are using ostree without a `/boot` partition and may be relying on this.  We should probably try to detect this dynamically?  Something like `test $(($(findmnt /boot | wc -l) > 0)) = 1` or something?  Or actually we already detect this inside `ostree-sysroot-deploy.c`; we could pass that via environment variable to this script?

Reply:
Actually I'm using OSTree without a ```/boot``` partition :)
So I understand this may break things for people that has ```/boot``` in a different partition, as in that case ```/boot/something/vmlinuz``` becomes just ```/something/vmlinuz``` when grub access the partition directly to load the kernel.

The suggestion of just checking if ```findmnt /boot``` reports something is not going to work because ```/boot``` is bind-mounted by [ostree-prepare-root in the initramfs](https://github.com/advancedtelematic/meta-updater/blob/afd18c7/recipes-sota/ostree-initrd/files/init.sh#L66) before doing the ```pivot_root``` and starting the actual system.

One possible idea is to check if ```/boot``` and ```/``` point to the same target device (```/dev/sda2``` in my case)

```
root@genericx86-64:~# findmnt /boot
TARGET SOURCE           FSTYPE OPTIONS
/boot  /dev/sda2[/boot] ext4   rw,relatime,data=ordered

root@genericx86-64:~# findmnt /    
TARGET SOURCE                                                                                                   FSTYPE OPTIONS
/      /dev/sda2[/ostree/deploy/poky/deploy/03dd817d1940a972a96256d952ba66dd8d5b7a43db5d19f048fd33d9c6c867b9.0] ext4   rw,relatime,data=ordered

root@genericx86-64:~# cat /proc/mounts | grep sda2
/dev/sda2 /sysroot ext4 rw,relatime,data=ordered 0 0
/dev/sda2 / ext4 rw,relatime,data=ordered 0 0
/dev/sda2 /boot ext4 rw,relatime,data=ordered 0 0
/dev/sda2 /usr ext4 ro,relatime,data=ordered 0 0
```

Reply:
Ah, I wasn't aware of that initramfs implementation.  Interesting.